Raivo/Features/DataImportFeature.swift

```diff
@@ -0,0 +1,111 @@
+        let iconValue: String
+        let secret: String
+        let issuer: String
+        let counter: String
+        let account: String
+        let iconType: String
+        let algorithm: String
+        let kind: String
+        let digits: String
+        let timer: String
+    }
+    
+    private func deleteFile(_ file: URL) {
+        try? FileManager.default.removeItem(at: file)
+    }
+    
+    private func deleteAllPasswords() {
+        if let realm = RealmHelper.shared.getRealm() {
+            let passwords = realm.objects(Password.self)
+            for password in passwords {
+                try? realm.write {
+                    password.deleted = true
+                }
+            }
+        }
+    }
+    
+    private func importNewPasswords(_ data: Data) {
+        let decoder = JSONDecoder()
+        do {
+            let jsonData = try decoder.decode([PasswordJson].self, from: data)
+            
+            for item in jsonData {
+                
+                // Construct a password
+                let password = Password()
+                password.id = password.getNewPrimaryKey()
+                password.issuer = item.issuer
+                password.account = item.account
+                password.iconType = item.iconType
+                password.iconValue = item.iconValue
+                password.secret = item.secret
+                password.algorithm = item.algorithm
+                password.digits = Int(item.digits) ?? 0
+                password.kind = item.kind
+                password.timer = Int(item.timer) ?? 0
+                password.counter = Int(item.counter) ?? 0
+                password.syncing = true
+                password.synced = false
+                password.pinned = Bool(item.pinned) ?? false
+                
+                // Store the password
+                if let realm = RealmHelper.shared.getRealm() {
+                    try realm.write {
+                        realm.add(password)
+                    }
+                }
+            }
+        } catch {
+            print("error importing passwords")
+        }
+    }
+    
+    public func importArchive(archiveFileURL: URL, withPassword password: String) -> (title: String, message: String) {
+        do {
+            // Unzip the file
+            let directory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+            try SSZipArchive.unzipFile(
+                atPath: archiveFileURL.path,
+                toDestination: directory.path,
+                overwrite: true,
+                password: password
+            )
+            
+            // Read the JSON data from the unzipped file
+            let filename = directory.appendingPathComponent("raivo-otp-export.json")
+            let data = try Data(contentsOf: filename)
+            deleteFile(filename)
+            
+            // Clean up old passwords
+            deleteAllPasswords()
```

Comment:
What are your thoughts on keeping all existing passwords, but overwriting those where the issuer and username match one from the backup? This way, users aren't left with duplicates each time they reimport. I also really like the feature proposed in #212 as a potential solution to this problem.

Reply:
I quite like the recycle bin, but it's also quite a lot of work to implement.

Anyhow, personally I quite like your suggestion to 'replace' tokens by username/issuer. However, I'm pretty sure there are less tech-savvy users out there who don't pay close attention and end up losing old tokens because of this. Therefore, I would suggest playing it safe and just warning users to manually delete old tokens first. Then the user cannot do anything wrong by accident.

Reply:
That makes sense: losing a token you can't get back is a way bigger problem than having to clean up a few duplicate/old OTPs. :)

All requested changes are now live in the latest commit. I've added an optional parameter called `shouldDeleteOldPasswords` in case we revisit this after implementing a recycle bin.